cmd/data-usage.go

```diff
@@ -48,40 +48,34 @@ func runDataUsageInfoUpdateRoutine() {
 		break
 	}
 
-	ctx := context.Background()
-
-	switch v := objAPI.(type) {
-	case *xlZones:
-		runDataUsageInfoForXLZones(ctx, v, GlobalServiceDoneCh)
-	case *FSObjects:
-		runDataUsageInfoForFS(ctx, v, GlobalServiceDoneCh)
-	default:
-		return
-	}
+	runDataUsageInfo(context.Background(), objAPI, GlobalServiceDoneCh)
 }
 
-func runDataUsageInfoForFS(ctx context.Context, fsObj *FSObjects, endCh <-chan struct{}) {
-	t := time.NewTicker(dataUsageCrawlInterval)
-	defer t.Stop()
-	for {
-		// Get data usage info of the FS Object
-		usageInfo := fsObj.crawlAndGetDataUsageInfo(ctx, endCh)
-		// Save the data usage in the disk
-		err := storeDataUsageInBackend(ctx, fsObj, usageInfo)
-		if err != nil {
-			logger.LogIf(ctx, err)
-		}
-		select {
-		case <-endCh:
-			return
-		// Wait until the next crawl interval
-		case <-t.C:
-		}
+func shouldCrawl(ctx context.Context, objAPI ObjectLayer) *time.Timer {
```

Comment:
Please document what the returned timer means.

If I read it correctly, this is what it does, so maybe:

```suggestion
// timeToNextCrawl returns the duration until next crawl should occur.
func timeToNextCrawl(ctx context.Context, objAPI ObjectLayer) time.Duration {
```

Returning a duration makes it more flexible. Also avoids needing to check for a nil timer.

Reply:
We will never return nil timer in this function,  returning duration is harder to handle with the select loop so I prefer to keep it this way.

Reply:
Fixed